lib/iris/tests/unit/experimental/ugrid/test_MeshCoord.py

```diff
@@ -431,5 +425,274 @@ def test_basic(self):
         self.assertArrayAllClose(auxcoord.bounds, meshcoord.bounds)
 
 
+class Test_MeshCoord__dataviews(tests.IrisTest):
+    """
+    Fuller testing of points and bounds calculations and behaviour.
+    Including connectivity missing-points (non-square faces).
+
+    """
+
+    def setUp(self):
+        self._make_test_meshcoord()
+
+    def _make_test_meshcoord(
+        self,
+        lazy_sources=False,
+        location="face",
+        inds_start_index=0,
+        inds_src_dim=0,
+        facenodes_changes=None,
+    ):
+        # Construct a miniature face-nodes mesh for testing.
+        # NOTE: we will make our connectivity arrays with standard
+        # start_index=0 and src_dim=0 :  We only adjust that (if required) when
+        # creating the actual connectivities.
+        face_nodes_array = np.array(
+            [
+                [0, 2, 1, 3],
+                [1, 3, 10, 13],
+                [2, 7, 9, 19],
+                [
+                    3,
+                    4,
+                    7,
+                    -1,
+                ],  # This one has a "missing" point (it's a triangle)
+                [8, 1, 7, 2],
+            ]
+        )
+        # Connectivity uses *masked* for missing points.
+        face_nodes_array = np.ma.masked_less(face_nodes_array, 0)
+        if facenodes_changes:
+            facenodes_changes = facenodes_changes.copy()
+            facenodes_changes.pop("n_extra_bad_points")
+            for indices, value in facenodes_changes.items():
+                face_nodes_array[indices] = value
+
+        # Construct a miniature edge-nodes mesh for testing.
+        edge_nodes_array = np.array([[0, 2], [1, 3], [1, 4], [3, 7]])
+        # Connectivity uses *masked* for missing points.
+        edge_nodes_array = np.ma.masked_less(edge_nodes_array, 0)
+
+        n_faces = face_nodes_array.shape[0]
+        n_edges = edge_nodes_array.shape[0]
+        n_nodes = int(face_nodes_array.max() + 1)
+        face_xs = 500.0 + np.arange(n_faces)
+        edge_xs = 600.0 + np.arange(n_edges)
+        node_xs = 100.0 + np.arange(n_nodes)
```

Comment:
@pp-mo You could make these magic numbers class attributes so that they can be referenced later in the tests e.g., 
```python
Test_MeshCoord__dataviews(...):
    FACE_OFFSET = 500.0
    EDGE_OFFSET = 600.0
    NODE_OFFSET = 100.0
```
Then use them as needed e.g., 
```python
node_xs = self.NODE_OFFSET + np.arange(n_nodes)
```
But more importantly, later is the actual tests i.e., see `_check_expected_bounds_values`, `test_edge_bounds` etc

Reply:
Done this.
I note that the face/edge values are not actually used.
Don't think that matters.